README.md

```diff
@@ -54,49 +67,161 @@ my_store = factory.get_hashstore(module_name, class_name, properties)
 # Store objects (.../[hashstore_path]/objects/)
 pid = "j.tao.1700.1"
 object = "/path/to/your/object.data"
-hash_address = my_store.store_object(pid, object)
-object_cid = hash_address.id
+object_metadata = my_store.store_object(pid, object)
+object_cid = object_metadata.id
 
 # Store metadata (.../[hashstore_path]/metadata/)
 # By default, storing metadata will use the given properties namespace `format_id`
 pid = "j.tao.1700.1"
 sysmeta = "/path/to/your/sysmeta/document.xml"
 metadata_cid = my_store.store_metadata(pid, sysmeta)
-```
 
-If you want to store other types of metadata, add an additional `format_id`.
-```py
+# If you want to store other types of metadata, add an additional `format_id`.
 pid = "j.tao.1700.1"
 metadata = "/path/to/your/metadata/document.json"
 format_id = "http://custom.metadata.com/json/type/v1.0"
 metadata_cid = my_store.store_metadata(pid, metadata, format_id)
+
+# ...
+```
+
+###### Working with objects (store, retrieve, delete)
+
+In HashStore, objects are first saved as temporary files while their content identifiers are calculated. Once the default hash algorithm list and their hashes are generated, objects are stored in their permanent location using the store's algorithm's corresponding hash value, the store depth and the store width. Lastly, reference files are created for the object so that they can be found and retrieved given an identifier (ex. persistent identifier (pid)). Note: Objects are also stored once and only once.
+
+By calling the various interface methods for  `store_object`, the calling app/client can validate, store and tag an object simultaneously if the relevant data is available. In the absence of an identfiier (ex. persistent identifier (pid)), `store_object` can be called to solely store an object. The client is then expected to call `verify_object` when the relevant metadata is available to confirm that the object has been stored as expected. If the object is determined to be invalid (via `verify_object`), it will be deleted. Lastly, to finalize this process of storing an object (to make the object discoverable), the client calls `tag_object`. In summary, there are two expected paths to store an object:
```

Comment:
I discussed `verify_object` with Jing and we determined that it should return a boolean value. If false is returned, the client is expected to delete the object directly. The ability to do this will be present once I refactor `delete_object` based on the new requirements described here: [Refactor deleteObject for new requirements](https://github.com/DataONEorg/hashstore-java/issues/57)

To your second question - yes, we will end up with a bunch of orphaned objects if a client calls `store_object` and forgets to `tag_object` (technically, you don't have to call `verify_object`...)